account_statement_import_sftp/components/account_statement_import_process.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Compassion CH
+# @author: Simon Gonzalez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.addons.component.core import Component
+
+
+class EdiBankStatementImportProcess(Component):
+    _name = "edi.input.process.bank.statement.import"
+    _usage = "input.process"
+    _backend_type = "bk_sftp"
+    _inherit = "edi.component.input.mixin"
+
+    def process(self):
+        statement_import = self.env["account.statement.import"].create(
+            [
+                {
+                    "statement_file": self.exchange_record.exchange_file,
+                    "statement_filename": self.exchange_record.exchange_filename,
+                }
+            ]
+        )
+        action = statement_import.import_file_button()
+        if not action:
+            raise ValueError(f"The import hasn't been achieved. {statement_import}")
+        statement = self.env["account.bank.statement"].browse(action.get("res_id"))
+        if not (statement.state and statement.state in ["posted", "open"]):
+            raise ValueError(
+                f"The statement {statement} opened\nHis state : {statement.state}"
```

Comment:
Raising an error means that the statement won't be created and won't be accessible after the error raised. It's thus useless to mention the state in my opinion. I don't really understand what is the need of this check. Maybe it would be better to raise the error at a lower-level for it to be more meaningful?
```suggestion
                _("The bank statement could not be validated.")
```